### PR TITLE
dispcalGUI: update to 3.9.8, change upstream.

### PR DESCRIPTION
**Pull request by kruceter**
dispcalGUI is not maintained by its original authors anymore.

The significant advantage of the fork is python 3 support which excludes python 2 from dependencies (related to #38229).

cc: @lemmi 

#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

**Comment by kruceter** [comment]
Found some flaws in the template, turning into draft until solved.

File: srcpkgs/dispcalGUI/template [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
-homepage="https://displaycal.net/"
-changelog="https://displaycal.net/CHANGES.html"
-distfiles="${SOURCEFORGE_SITE}/dispcalgui/${_name}-${version}.tar.gz"
-checksum=7c34dfbd9f66f24f1d4c88de4a5a0de688aad719f095874b6259637d30893bea
+homepage="https://github.com/eoyilmaz/displaycal-py3"
+changelog="https://github.com/eoyilmaz/displaycal-py3/releases/tag/${version}"
+distfiles="https://github.com/eoyilmaz/displaycal-py3/archive/${version}.tar.gz"
+checksum=a9e8bba02f9bdf8df506f0dcbbea4eae48795001cb81ea5968fbfc41f4b72e1d
+# requires running X11 server and dbus-daemon
+make_check=no
```
Comment (paper42) [inline srcpkgs/dispcalGUI/template:19]: would this help?
```
checkdepends="xvfb-run dbus"
make_check_pre="xvfb-run dbus-run-session"
```

File: srcpkgs/dispcalGUI/template [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
+
+post_patch() {
+	# set the hardcoded version of python 3 to the latest one
+	sed -i "s|^\(py_maxversion =\).*|\1 (${py3_ver%.*}, ${py3_ver#*.})|" DisplayCAL/meta.py
```
Comment (paper42) [inline srcpkgs/dispcalGUI/template:23]: replace all seds with vsed to make sure we know when they stop changing anything

**Review by paper42 — COMMENTED** [review]
Could you try upstreaming the patches? When that's done, we can consider merging this, but right now this package would be very hard to maintain.

File: srcpkgs/dispcalGUI/template (hunk 1 above)
Comment (kruceter) [reply srcpkgs/dispcalGUI/template:19]: It did solve the issue with X11, but dbus is persistent as ever.

[dispcalgui.txt](https://github.com/void-linux/void-packages/files/9996060/dispcalgui.txt)

**Comment by kruceter** [comment]
> Could you try upstreaming the patches?

I will try to upstream them.

File: srcpkgs/dispcalGUI/template (hunk 1 above)
Comment (paper42) [reply srcpkgs/dispcalGUI/template:19]: I would guess it needs a running system dbus service, so just adjust the comment.

**Comment by kruceter** [comment]
Unfortunately, my efforts to fix upstream issues are futile.

With the new upstream being as it is at present time, I find the idea of porting this package to be an eternal game of Whac-A-Mole.

**Comment by lemmi** [comment]
@classabbyamp pointed out the new release of `dispcalgui-py3`: https://github.com/eoyilmaz/displaycal-py3/releases/tag/3.9.11
Maybe this is worth another look and if it works, it's probably best to add a new `dispcalgui-py3` package that ships with `replaces=dispcalgui`.
@kruceter interested in giving this another go?

**Pull request closed without merging**